docking/rigorous_work.py

```diff
@@ -375,46 +335,50 @@ def do_deletion(x0, v0, combined_bps, combined_masses, box, guest_name, leg_type
 
     ctxt = custom_ops.Context(x0, v0, box, intg, u_impls)
 
-    subsample_freq = 2
-    du_dl_obs = custom_ops.FullPartialUPartialLambda(u_impls, subsample_freq)
-    ctxt.add_observable(du_dl_obs)
+
+    # du_dl_obs = custom_ops.FullPartialUPartialLambda(u_impls, subsample_freq)
+    # ctxt.add_observable(du_dl_obs)
 
     deletion_lambda_schedule = np.linspace(
         MIN_LAMBDA, DELETION_MAX_LAMBDA, TRANSITION_STEPS
     )
 
     calc_work = True
 
-    for step, lamb in enumerate(deletion_lambda_schedule):
-        ctxt.step(lamb)
-        if step % 100 == 0:
-            report.report_step(
-                ctxt,
-                step,
-                lamb,
-                box,
-                combined_bps,
-                u_impls,
-                guest_name,
-                TRANSITION_STEPS,
-                f"{leg_type.upper()}_DELETION",
-            )
-        if step in (0, int(TRANSITION_STEPS/2), TRANSITION_STEPS-1):
-            if report.too_much_force(ctxt, lamb, box, combined_bps, u_impls):
-                calc_work = False
-                return
+    subsample_freq = 1
+    full_du_dls = ctxt.multiple_steps(deletion_lambda_schedule, subsample_freq)
```

Comment:
Can/should we use the ctxt.multiple_steps() for the insertion & equilibration, too?